### PR TITLE
This is to fix issue: https://github.com/alibaba/RocketMQ/issues/359

### PR DESCRIPTION
**Pull request by lizhanhui**
https://github.com/alibaba/RocketMQ/issues/359

### CODE REVIEW

**Comment by vongosling** [comment]
thank you pull request. @zhouxinyu

**Comment by lizhanhui** [comment]
Try this unit test:    



    @Test
    public void testNonAsciiGroupName() {

        String nonAsciiGroupName = "中文";

        SubscriptionGroupManager subscriptionGroupManager = new SubscriptionGroupManager();
        SubscriptionGroupConfig nonAsciiGroup = new SubscriptionGroupConfig();
        nonAsciiGroup.setGroupName(nonAsciiGroupName);
        subscriptionGroupManager.updateSubscriptionGroupConfig(nonAsciiGroup);
        subscriptionGroupManager.persist();
        Assert.assertNotNull(subscriptionGroupManager.findSubscriptionGroupConfig(nonAsciiGroupName));

        BufferedReader bufferedReader = null;

        SubscriptionGroupManager manager = new SubscriptionGroupManager();
        manager.load();

        try {
            bufferedReader = new BufferedReader(new FileReader(new File(manager.configFilePath())));
            boolean found = false;
            String line = "";
            while ((line = bufferedReader.readLine()) != null) {
                if (line.contains(nonAsciiGroupName)) {
                    found = true;
                    break;
                }
            }
            Assert.assertTrue(found);
        } catch (FileNotFoundException e) {
            Assert.fail("File should exist");
        } catch (IOException e) {
            Assert.fail("Unexpected IO exception");
        } finally {
            if (null != bufferedReader) {
                try {
                    bufferedReader.close();
                } catch (IOException ignore) {
                }
            }
        }

        try {
            SubscriptionGroupConfig shouldExist = manager.findSubscriptionGroupConfig(nonAsciiGroupName);
            Assert.assertNotNull(shouldExist);
        } catch (NullPointerException e) {
            Assert.fail("Bug reproduced: NPE is thrown as specified non-ascii group is not found");
        }
    }

**Comment by lizhanhui** [comment]
@vongosling  Who is @zhouxinyu? 

The involved issue was first reported by @Jaskey and then followed by @gaoyf

IMO, these two guys should be thanked.

**Comment by Jaskey** [comment]
@vongosling does this pr do fix the issue without any side effect?

File: rocketmq-broker/src/main/java/com/alibaba/rocketmq/broker/slave/SlaveSynchronize.java [hunk 1]
```diff
@@ -72,13 +74,31 @@ private void syncTopicConfig() {
...
                             .putAll(topicWrapper.getTopicConfigTable());
                     this.brokerController.getTopicConfigManager().persist();
 
+                    // delete consume queues of unused topics.
+                    if (!unusedTopics.isEmpty()) {
+                        StringBuilder stringBuilder = new StringBuilder();
+                        for (String topic : unusedTopics) {
+                            stringBuilder.append(topic).append(", ");
+                        }
+                        log.info("Consume queues of topic: {} are to be cleaned", stringBuilder.substring(0, stringBuilder.length() - 2));
```
Comment (Jaskey) [inline rocketmq-broker/src/main/java/com/alibaba/rocketmq/broker/slave/SlaveSynchronize.java]: to keep code clean, directly log with Set's to String is enough , no need for a for loop to build a StringBuilder.

File: rocketmq-broker/src/main/java/com/alibaba/rocketmq/broker/slave/SlaveSynchronize.java [hunk 2]
```diff
@@ -72,13 +74,31 @@ private void syncTopicConfig() {
                 if (!this.brokerController.getTopicConfigManager().getDataVersion()
                         .equals(topicWrapper.getDataVersion())) {
 
+                    // figure out unused topics.
+                    Set<String> unusedTopics = new HashSet<String>();
```
Comment (Jaskey) [inline rocketmq-broker/src/main/java/com/alibaba/rocketmq/broker/slave/SlaveSynchronize.java]: To keep log more readable, tree set will be more desirable?

**Pull request closed without merging**